src/data_parser.py

```diff
@@ -1,8 +1,10 @@
-import data_fetcher
 import time
+from typing import Dict, List
+
+import data_fetcher
 
 
-def get_subjects(year: int):
+def get_subjects(year: int) -> Dict[str, List[Dict[str, str]]]:
```

Comment:
Using `dict` and `list` is fine as of Python 3.7 or 3.8. The `typing` module isn't needed